pkg/virtctl/convert-machine-type/convert-machine-type-job/update_machine_type.go

```diff
@@ -0,0 +1,152 @@
+package convertmachinetypejob
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	k8sv1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	k6tv1 "kubevirt.io/api/core/v1"
+)
+
+// using these as consts allows us to easily modify the program to update as newer versions are released
+// we generally want to be updating the machine types to the most recent version
+const (
+	LatestMachineTypeVersion           = "rhel9.2.0"
+	MinimumSupportedMachineTypeVersion = "rhel9.0.0"
+)
+
+var (
+	// by default, update machine type across all namespaces
+	namespace     = k8sv1.NamespaceAll
+	labelSelector = ""
+
+	// by default, should require manual restarting of VMIs
+	restartNow = false
+)
+
+func IsMachineTypeUpdated(machineType string) (isUpdated bool, updatedMachineType string, err error) {
+
+	// if the machine type is "q35", we want the VM to behave as though its machine type has been updated
+	if machineType == "q35" {
+		return true, machineType, nil
+	}
+
+	machineTypeSubstrings := strings.Split(machineType, "-")
+
+	if len(machineTypeSubstrings) != 3 {
+		return false, machineType, fmt.Errorf("invalid machine type: %s", machineType)
+	}
+
+	if len(machineTypeSubstrings) == 3 {
+		version := machineTypeSubstrings[2]
+		if strings.Contains(version, "rhel") && version < MinimumSupportedMachineTypeVersion {
+			machineTypeSubstrings[2] = LatestMachineTypeVersion
+			machineType = strings.Join(machineTypeSubstrings, "-")
+			return true, machineType, nil
+		}
+	}
+	return false, machineType, nil
+}
+
+func (c *JobController) UpdateMachineTypes() error {
+	listOpts := &k8sv1.ListOptions{}
+	if labelSelector != "" {
+		listOpts = &k8sv1.ListOptions{
+			LabelSelector: labelSelector,
+		}
+	}
+	vmList, err := c.VirtClient.VirtualMachine(namespace).List(context.Background(), listOpts)
+
+	if err != nil {
+		return err
+	}
+
+	for _, vm := range vmList.Items {
+		machineType := vm.Spec.Template.Spec.Domain.Machine.Type
+		needsUpdate, machineType, err := IsMachineTypeUpdated(machineType)
+
+		if err != nil {
+			fmt.Print(err)
+			continue
+		}
+
+		// skip VM if its machine type is supported
+		if !needsUpdate {
+			continue
+		}
+
+		if machineType != "q35" {
```

Comment:
What if `needsUpdate == true` but `machineType == "q35"`? Even if it's an in-code assumption that's logically not possible, I find this a bit confusing.

Do we really need this `if`? If we don't need update, we would `continue` above anyhow, so why not execute it unconditionally?